lib/compiler_rt/udivmodei4.zig

```diff
@@ -0,0 +1,121 @@
+const std = @import("std");
+const common = @import("common.zig");
+const shr = std.math.shr;
+const shl = std.math.shl;
+
+const max_limbs = std.math.divCeil(usize, 65535, 32) catch unreachable; // max supported type is u65535
+
+comptime {
+    @export(__udivei4, .{ .name = "__udivei4", .linkage = common.linkage });
+    @export(__umodei4, .{ .name = "__umodei4", .linkage = common.linkage });
+}
+
+fn divmod(q: ?[]u32, r: ?[]u32, u: []const u32, v: []const u32) !void {
```

Comment:
Might be nice to mention that this uses `Knuth's Algorithm D` for later review.
Please add `@setRuntimeSafe(is_test)` (https://ziglang.org/documentation/master/#toc-setRuntimeSafety).